test/unit/test_jinja.py

```diff
@@ -413,3 +415,61 @@ def test_if_endfor_newlines(self):
 '''
 
 
+native_expected_behaviors = [
+    # strings
+    ('''foo: bar''', 'bar'),
+    ('''foo: "bar"''', 'bar'),
+    ('''foo: "'bar'"''', "'bar'"),
+    ("""foo: '"bar"'""", '"bar"'),
+    # ints
+    ('''foo: 1''', 1),
+    ('''foo: "1"''', 1),
+    ('''foo: "'1'"''', "'1'"),
+    ('''foo: "{{ 1 }}"''', 1),
+    ('''foo: "{{ '1' }}"''', 1),
+    ('''foo: "'{{ 1 }}'"''', "'1'"),
+    ('''foo: "'{{ '1' }}'"''', "'1'"),
+    ('''foo: "{{ 1 | as_text }}"''', '1'),
+    ('''foo: "{{ '1' | as_text }}"''', '1'),
+    # booleans.
+    # Note the discrepancy with true vs True: `true` is recognized by jinja but
+    # not literal_eval, but `True` is recognized by ast.literal_eval.
+    # For extra fun, yaml recognizes both.
+    ('''foo: "{{ true }}"''', True),
+    ('''foo: "{{ 'true' }}"''', 'true'),
+    ('''foo: "'{{ true }}'"''', "'True'"),
+    ('''foo: "{{ true | as_text }}"''', "True"),  # true -> boolean True -> text -> str(True) -> 'True'
```

Comment:
this one is surprising, but i guess that's what we should expect if `true == True`!

Reply:
I put that comment there because my initial reaction to this behavior was that it just had to be wrong 😆